File: 03_hashing/03_01_count_frequency_of_all_elements/count_frequency_of_elements.js
function countFrequencyOfElements(arr) {
    const dictionary = {};
    for (let i = 0; i < arr.length; i++) {
        if (dictionary[arr[i]]) {
            dictionary[arr[i]]++;
        } else {
            dictionary[arr[i]] = 1;
        }
    }
    return dictionary;
}

const arr = [10, 5, 10, 15, 10, 5];
const result = countFrequencyOfElements(arr);

for (const [key, value] of Object.entries(result)) {
    console.log(key + '-> ' + value);
}

File: 03_hashing/03_01_count_frequency_of_all_elements/count_frequency_of_elements.py
def count_frequency_of_elements(arr: list, n: int, dictionary: dict):
    for i in range(n):
        if arr[i] in dictionary:
            dictionary[arr[i]] += 1
        else:
            dictionary[arr[i]] = 1
    for x in dictionary:
        print(x, dictionary[x])
        
        
arr = [10, 5, 10, 15, 10, 5]
n = len(arr)
result = {}
count_frequency_of_elements(arr, n, result)
print(result)